src/commands/tail/mod.rs

```diff
@@ -52,18 +53,31 @@ async fn run_cloudflared_start_server(
     }
 }
 
+async fn start_log_collection_http_server() -> Result<(), failure::Error> {
```

Comment:
```suggestion
async fn start_log_collection_http_server() -> ! {
```

if i understand correctly this will never return (unless we start handling sigint/sigkill)